sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/binaryfile/BinaryFileFormat.scala

```diff
@@ -115,6 +118,11 @@ class BinaryFileFormat extends FileFormat with DataSourceRegister {
             case (MODIFICATION_TIME, i) =>
               writer.write(i, DateTimeUtils.fromMillis(status.getModificationTime))
             case (CONTENT, i) =>
+              if (status.getLen > maxLength) {
```

Comment:
I think we can move this to line 113.

Reply:
I don't get it. The conf is to prevent reading very large files that we are sure about failures. User can still use the data source if they don't need `content`.

Reply:
I see. I am actually OK with either way.